packages/flutter/lib/src/material/material.dart

```diff
@@ -38,6 +38,27 @@ enum MaterialType {
   transparency
 }
 
+/// Defines the appearance of an ink splash displayed by [InkResponse] or
+/// [InkWell].
+///
+/// See also:
+///
+///   * [ThemeData.splashType], which defines the splash type for a Material
+///     [Theme].
+enum InkSplashType {
```

Comment:
what i wouldn't give for metaclasses.

How about we put a static function on the two classes, each implementing the same signature as the constructors and each just redirecting to the respective constructor, and then instead of this enum we point to one of those functions?

Then if someone wants to make a third kind of splash, it's easy to do.

It's ugly as heck, but InkSplashType is BoxShape all over again...

Reply:
I defined an `InteractiveInkFeatureFactory` (yow) abstract class that just has a create method. It's the value of `InkSplash.splashFactory` and `InkRipple.splashFactory`